openpharma/staged.dependencies
# Add renv lockfiles into object and use renv::install for external deps if inside renv environment

Closes: https://github.com/openpharma/staged.dependencies/issues/147 and closes #146 
<img width="442" alt="image" src="https://user-images.githubusercontent.com/15201933/143031158-2c9f3e08-b619-473d-a171-f770f93071c8.png">

To test #147:

```
x <- dependency_table("openpharma/stageddeps.house", "repo@host", ref = "renv@main")
print(x$renv_files$stageddeps.house)
print(x$renv_files$stageddeps.food)
# test a profile which only exists on stageddeps.house:
x <- dependency_table("openpharma/stageddeps.house", "repo@host", ref = "profiles@renv@main", renv_profile = "example")
print(x$renv_files$stageddeps.house)
print(x$renv_files$stageddeps.food)
```


To test #146 

First run - outside of an renv environment:

```
x <- dependency_table("openpharma/stageddeps.house", "repo@host", ref = "renv@main") # could use "main" 
install_deps(x)
```

Then create an renv environment and make sure Bioconductor repos are in options("repos") and run the same command and you'll see the dependencies coming from the cache (or being installed into the cache if they are not already in the cache)
